src/tests.rs

```diff
@@ -988,3 +986,109 @@ fn add_where_item() {
     assert_quote_snapshot!(basic_type);
     assert_quote_snapshot!(type_with_args);
 }
+
+// =================
+// IMPL DECLARATIONS
+// =================
+
+#[test]
+fn parse_impl_inherent() {
+    let expr = quote! {
+        impl MyStruct {
+            fn new(i: i32, b: bool) -> Self {
+                Self {}
+            }
+
+            #[attr]
+            pub(crate) fn set_value(&mut self, s: String) {}
+
+            pub const CONSTANT: i8 = 24 + 7;
+        }
+    };
+
+    let impl_decl = parse_declaration_checked(expr);
+    assert_debug_snapshot!(impl_decl);
```

Comment:
Have you considered using `assert_quote_snapshot` ? (defined higher in the file)